### PR TITLE
Sui deploy demos and protection

### PR DESCRIPTION
**Pull request by davidbarna**
## Description

* [x] Upgrade warning as a major version is about to be published.
* [x] Add --auth option (default) to avoid deployments to be accessible by search engines.
* [x] Add --branch for easier branch deployments (demos in pull requests)

### CODE REVIEW

File: packages/sui-deploy/README.md [hunk 1]
```diff
@@ -60,15 +101,16 @@ $ sui-deploy spa --help
   "version": "1.0.0",
   "scripts": {
     "build": "sui-bundler build -C",
-    "deploy": "sui-deploy test-project --now"
+    "deploy": "sui-deploy test-project --now -a 'my-user:my-password'"
```
Comment (carlosvillu) [inline packages/sui-deploy/README.md]: Which is the default public folder?!

Comment (davidbarna) [reply packages/sui-deploy/README.md]: https://github.com/SUI-Components/sui/pull/276/files#diff-304d634116a986f8a14b512c69a22832R23

**Comment by carlesnunez** [comment]
If sui-deploy have at this moment the only purpose of deploy SPA's into now via the --now command I asume that we are setting the --now command as maybe in the future we plan to add another providers.

In addition, for me, the naming of the package is a bit confusing as I asume that this package will help me more in the deployment of my application than in the deployment of demo's... so, just to have it clear, what's exactly the roadmap or plan of this package in the near future?

**Comment by davidbarna** [comment]
@carlesnunez combining sui-deploy with sui-ssr (the generates a standalone folder that you can upload anywhere) we'll be able to use sui-deploy for production deployments. It's part of the roadmap.

**Comment by carlesnunez** [comment]
This sounds truly f**ng amazing!! Thank you guys! 👍

**Pull request merged**